be/dtm0_log.h

```diff
@@ -0,0 +1,236 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+#pragma once
+#ifndef __MOTR_BE_DTM0_LOG_H__
+#define __MOTR_BE_DTM0_LOG_H__
+
+/**
+ *  @page dtm0 log implementation
+ *
+ *  @section Overview
+ *  DTM0 log module will be working on incoming message request, the
+ *  goal of this module is to track journaling of incoming message either on
+ *  persistent or volatile memory based on whether logging is happening on
+ *  participant or on originator, so that in the phase failure consistency of
+ *  failed participant can be restored by iterating over logged journal to
+ *  decide which of the logged messages needs to be sent as a redo request.
+ *
+ *  During normal operation when every participant and originator is online
+ *  every transaction will be logged and at any point state of transaction can
+ *  be any of the following,
+ *
+ *	enum m0_dtm0_log_participant_state {
+ *		M0_DTML_STATE_UNKNOWN,
+ *		M0_DTML_STATE_SENT,
+ *		M0_DTML_STATE_EXECUTED,
+ *		M0_DTML_STATE_PERSISTENT,
+ *	};
+ *
+ *  Every participant maintains the journal record corresponds to each
+ *  transaction and it can be described by "struct m0_dtm0_log_record" which
+ *  will be stored in persistent storage. Basically this record will maintain
+ *  txr id, state of transaction on all the participant and payload.
+ *
+ *  The state of transaction for which participant is adding log can be
+ *  M0_DTML_STATE_EXECUTED or M0_DTML_STATE_PERSISTENT. For the same transaction
+ *  state of remote participant can be either M0_DTML_STATE_UNKNOWN or
+ *  M0_DTML_STATE_PERSISTENT, this is because remote participant will only send
+ *  state of it's transaction to the other participant when transaction become
+ *  persistent.
+ *
+ *  Originator also maintains the same transaction record in volatile memory.
+ *  Originator is expecting to get replys from participant when transaction on
+ *  participant is M0_DTML_STATE_EXECUTED or M0_DTML_STATE_PERSISTENT.
+ *  On originator the state of the transaction can be M0_DTML_STATE_UNKNOWN to
+ *  M0_DTML_STATE_PERSISTENT for each participant. O
+ *
+ *  During recovery operation of any of the participant, rest of the participant
+ *  and originator will iterate over the logged journal and extract the state
+ *  of each transation for participant under recovery from logged information
+ *  and will send redo request for those transaction which  are not
+ *  M0_DTML_STATE_PERSISTENT on participant being recovered.
+ *
+ *  Upon receiving redo request participant under recovery will log the state
+ *  of same transaction of remote participant in persistent store.
+ *
+ * @section Usecases
+ *
+ * 1. When transaction successfully executed on participant, DTM0 log will be
```

Comment:
Suggestion: "DTM0 log entry" or "DTM0 log record" instead of just "DTM0 log".

Reply:
Addressed at 4e0d3688b2cc72748d9c1ed8df1ddeb8a2319fba